Google.PowerShell/Common/GoogleAnalyticsReporter.cs

```diff
@@ -0,0 +1,224 @@
+            var hitData = new Dictionary<string, string>(_baseHitData)
+            {
+                { HitTypeParam, EventTypeValue },
+                { EventCategoryParam, category },
+                { EventActionParam, action },
+            };
+            if (label != null)
+            {
+                hitData[EventLabelParam] = label;
+            }
+            if (value != null)
+            {
+                hitData[EventValueParam] = value.ToString();
+            }
+            SendHitData(hitData);
+        }
+
+        /// <summary>
+        /// Reports a window view.
+        /// </summary>
+        /// <param name="name">The name of the window. Must not be null.</param>
+        public void ReportScreen(string name)
+        {
+            Preconditions.CheckNotNull(name, nameof(name));
+
+            var hitData = new Dictionary<string, string>(_baseHitData)
+            {
+                { HitTypeParam, ScreenViewValue },
+                { ScreenNameParam, name },
+            };
+            SendHitData(hitData);
+        }
+
+        /// <summary>
+        /// Reports that the session is starting.
+        /// </summary>
+        public void ReportStartSession()
+        {
+            var hitData = new Dictionary<string, string>(_baseHitData)
+            {
+                { HitTypeParam,EventTypeValue },
+                { SessionControlParam, SessionStartValue }
+            };
+            SendHitData(hitData);
+        }
+
+        /// <summary>
+        /// Reports that the session is ending.
+        /// </summary>
+        public void ReportEndSession()
+        {
+            var hitData = new Dictionary<string, string>(_baseHitData)
+            {
+                { HitTypeParam, EventTypeValue },
+                { SessionControlParam, SessionEndValue }
+            };
+            SendHitData(hitData);
+        }
+
+        /// <summary>
+        /// Constructs the dictionary with the common parameters that all requests must
+        /// have.
+        /// </summary>
+        /// <returns>Dictionary with the parameters for the report request.</returns>
+        private Dictionary<string, string> MakeBaseHitData()
+        {
+            var result = new Dictionary<string, string>
+            {
+                { VersionParam, VersionValue },
+                { PropertyIdParam, PropertyId },
+                { ClientIdParam, ClientId },
+                { AppNameParam, ApplicationName },
+            };
+            if (ApplicationVersion != null)
+            {
+                result.Add(AppVersionParam, ApplicationVersion);
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Sends the hit data to the server.
+        /// </summary>
+        /// <param name="hitData">The hit data to be sent.</param>
+        private async void SendHitData(Dictionary<string, string> hitData)
+        {
+            using (var client = new HttpClient())
```

Comment:
I think HttpClient is supposed to be reused (https://www.asp.net/web-api/overview/advanced/calling-a-web-api-from-a-net-client) so maybe we shouldn't create a new instance of HttpClient everytime.